### ISSUE TITLE
Support for Hostname Attribute for Media Streaming Endpoint Resource

### ISSUE THREAD

**Issue by sam-mundle** [terraform-providers/terraform-provider-azurerm#10458]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

When trying to use a Standard Microsoft CDN endpoint for a streaming endpoint rather than one of the 3 native options (`StandardVerizon`, `PremiumVerizon`, `StandardAkamai`) it would be nice to have an attribute `host_name` for `azurerm_media_streaming_endpoint` so that we can use that attribute for the origin in an `azurerm_cdn_endpoint` resource. This is necessary because after speaking with someone on the Azure team, they informed me that the origin schema for streaming endpoints has changed in the past and could change again. Having only an attribute change when redeploying our CDN to point at a new Origin hostname would be far better than building the hostname URL string as we are right now.

TL;DR It would be great to have have a `host_name` attribute for the `azurerm_media_streaming_endpoint` resource

### New or Affected Resource(s)

* azurerm_media_streaming_endpoint

### Potential Terraform Configuration

```hcl
resource "azurerm_cdn_endpoint" "example" {
  name                = "example"
  profile_name        = azurerm_cdn_profile.example.name
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  origin {
    name      = "example"
    host_name = azurerm_media_streaming_endpoint.host_name <------ (this would be the changed implementation)
  }
}
```

### References

* #9537
